geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/procedures/RollWriteAheadLog.scala

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis.dialect
+package procedures
+
+import org.locationtech.geomesa.gt.partition.postgis.dialect.tables.WriteAheadTable
+
+import java.util.Locale
+
+/**
+ * Moves the current write ahead table to a sequential name, and creates a new write ahead table
+ * to accept writes going forward
```

Comment:
At a high-level, how are incoming writes handled during a roll?

Reply:
they'll be queued up/paused due to the locking required by each, but only for a very brief time period that isn't really noticeable.